src/scaffold/entry_type/integrity.rs

```diff
@@ -846,54 +855,62 @@ fn add_entry_type_to_validation_arms(
+                                                        let original_record = must_get_valid_record(original_action_hash)?;
+                                                        let original_action = original_record.action().clone();
+                                                        let original_action = EntryCreationAction::try_from(original_action)
+                                                            .map_err(|e| wasm_error!(e.to_string()))?;
+                                                        let app_entry_type = match original_action.entry_type() {
+                                                            EntryType::App(app_entry_type) => app_entry_type,
+                                                            _ => {
+                                                                return Ok(ValidateCallbackResult::Valid);
+                                                            }
+                                                        };
+                                                        let entry = match original_record.entry().as_option() {
+                                                            Some(entry) => entry,
+                                                            None => {
+                                                                if original_action.entry_type().visibility().is_public() {
+                                                                    return Ok(ValidateCallbackResult::Invalid(
+                                                                        "Original record for a delete of a public entry must contain an entry"
+                                                                            .to_string(),
+                                                                    ));
+                                                                } else {
+                                                                    return Ok(ValidateCallbackResult::Valid);
                                                                 }
                                                             }
-                                                        }
-                                                    }
+                                                        };
+                                                        let original_app_entry = match EntryTypes::deserialize_from_type(
+                                                            app_entry_type.zome_index,
+                                                            app_entry_type.entry_index,
+                                                            entry,
+                                                        )? {
+                                                            Some(app_entry) => app_entry,
+                                                            None => {
```

Comment:
Actually, when looking at the `OpDelete` from before the breaking change, e.g. [here](https://docs.rs/hdi/0.4.0-beta-dev.35/hdi/flat_op/enum.OpDelete.html) I see that there are also other types of entry, namely `Agent`, `CapClaim`, CapClaim` and `PrivateEntry`. Private entry is already covered above but the others aren't. I see that the previous scaffolding all accepted them as valid which also seems problematic to me. But considering them all as invalid like here  be at least as problematic. I just don't understand enough about under which circumstances such other types of entries would make it into validation...
@ThetaSinner do you have any insights here?

Reply:
The `OpDelete` used to be an enum [before](https://docs.rs/hdi/0.4.0-beta-dev.35/hdi/flat_op/enum.OpDelete.html), now it's a [struct](https://docs.rs/hdi/0.4.0-beta-dev.36/hdi/flat_op/struct.OpDelete.html), I assume all the other variants were composed into one a single `OpDelete` struct as per this change https://github.com/holochain/holochain/pull/3627/files#diff-e0f69b27e830196a207906bed69310346863f12e983f2e9ea3acb502e8a12640R149

Reply:
@ThetaSinner  I am also very curious about the rationale of removing the generic type `ET` parameter from `OpDelete`, apart from turning the previous enum into a struct

Reply:
Yes right but this seems to make it rather cumbersome to figure out the type of EntryDeletion we're validating. And I'm unclear on how to properly validate types of entries other than ones defined in `EntryTypes`.
@guillemcordoba do you maybe have some thoughts on that? What your rationale was on accepting deletes of all other types of entries than the ones defined in `EntryTypes` as valid?

Reply:
The best person to ask about this would be @jost-s, who made the change. 

As for the general discussion about what the validation rules should do here. Ideally not too much really, it's up to the app author what is and isn't valid and they should be writing tests as they go. Any common validation that should always be run should be happening in sys validation anyway really. I haven't had a chance to look too closely at this yet but is this already doing too much?

Reply:
You’re correct that without the original entry, it’s unclear what type the deleted entry was. This requires additional code to determine the type. It appears that this is similar to the behavior observed with `OpRecord::DeleteEntry`, similar rationale as well I suppose?

Reply:
Yes, generally for `Op` the original action and entry were removed. This is a performance improvement with the assumption that you would rarely need the original entry when validating updates and deletes. If you do need it in your happ, you can `must_get_entry` or `must_get_action` for the action.

Reply:
> The rationale was that it's unlikely you'd want to check the original entry that you're deleting in validation. What do you think?
I don't think it's that uncommon to require the `EntryType`. If you have 4 different `EntryType`s and one of them is not allowed to be deleted, then you need to match on the entry type.

Reply:
This is not saying that this optimization is not justified...but it seems difficult to make use of it in the scaffolded boilerplate.

Reply:
which would mean that the optimization would not actually make it into scaffolded apps in practice...

Reply:
It's an optimization in Holochain when preparing `Op`s in the app validation workflow.

> This is not saying that this optimization is not justified...but it seems difficult to make use of it in the scaffolded boilerplate.
In the boilerplate you need to do what `flattened` used to do previously, if you want to check different ones of the EntryTypes. `must_get_entry`, then match against the `EntryType` (app, capgrant, ...) and match against the app's entry types.